compiler/rustc_passes/src/check_const.rs

```diff
@@ -199,7 +199,7 @@ impl<'tcx> Visitor<'tcx> for CheckConstVisitor<'tcx> {
             ..
         }) = item.kind
         {
-            let def_id = trait_ref.trait_def_id().unwrap();
+            let Some(def_id) = trait_ref.trait_def_id() else { return; };
```

Comment:
Do we want to return here or break out of the block so that `intravisit::walk_item` still runs?